tracy-client/src/lib.rs

```diff
@@ -148,6 +148,23 @@ impl Client {
             internal::set_thread_name(name.as_ptr().cast());
         }
     }
+
+    /// Set the current thread name according to Rust's [`Thread::name()`].
+    ///
+    /// This is useful if the thread name is already set with Rust's standard library APIs, and you
+    /// want to copy the name to Tracy. Does nothing if the current thread is not named according
+    /// to [`Thread::name()`].
+    ///
+    /// [`Thread::name()`]: std::thread::Thread::name
+    pub fn set_thread_name_from_std(&self) {
+        #[cfg(feature = "enable")]
+        {
+            if let Some(name) = std::thread::current().name() {
+                // This will not panic because Rust disallows thread names with null bytes.
+                self.set_thread_name(name);
+            }
+        }
+    }
```

Comment:
This is trivial for downstream code to write themselves; is it really worth having a method for it?

Reply:
I put the method here to raise some awareness of the Rust thread naming API, as `set_thread_name` alone may imply thread naming being some Tracy-specific feature.

Reply:
Could that be solved by pure documentation instead?